core/metacling/src/TCling.cxx

```diff
@@ -2775,6 +2775,25 @@ void TCling::EndOfLineAction()
    ProcessLineSynch(fantomline);
 }
 
+// This static function is a hop of TCling::IsLibraryLoaded, which is taking a lock and calling
+// into this function. This is because we wanted to avoid a duplication in TCling::IsLoaded, which
+// was already taking a lock.
+static Bool_t s_IsLibraryLoaded(const char* libname, cling::Interpreter* fInterpreter)
+{
+   // Check shared library.
+   TString tLibName(libname);
+   const char *fullLibName = gSystem->FindDynamicLibrary(tLibName, kTRUE);
```

Comment:
or (IMHO nicer):
```
if (gSystem->FindDynamicLibrary(tLibName, kTRUE))
   return fInterpreter->getDynamicLibraryManager()->isLibraryLoaded(tLibName);
```

Reply:
I think we have to do FindDynamicLibrary in order to pass isLibraryLoaded a fullpath, no?
`bool isLibraryLoaded(llvm::StringRef fullPath) const;`

Reply:
Check the `FindDynamicLibrary` you are calling! It's modifying the `tLibName` argument, and returns a `const char*` to it.